reactor-core/src/main/java/reactor/core/publisher/Sinks.java

```diff
@@ -167,15 +167,19 @@ public void orThrowWithCause(Throwable cause) {
 		final Emission reason;
 
 		public EmissionException(Emission reason) {
-			super("Sink emission failed with " + reason);
-			this.reason = reason;
+			this(reason, "Sink emission failed with " + reason);
 		}
 
 		public EmissionException(Throwable cause, Emission reason) {
 			super("Sink emission failed with " + reason, cause);
 			this.reason = reason;
 		}
 
+		public EmissionException(Emission reason, String message) {
```

Comment:
we need either to make it consistent with the above constructor or open a separate PR to swap the parameters of the above constructor, wdyt? (the later would be breaking between RC1 and RC2, but probably a more meaningful order)

Reply:
will do in a follow up PR